### DISCUSSION TITLE
Is there a way to safely change the world type from Vanilla to Forge?

### DISCUSSION THREAD

**Discussion by Tingweiftw** [itzg/docker-minecraft-server#2337] [Q&A] [answered]
I successfully hosted a vanilla server 1.20.1 on my raspberry pi as a docker container via Portainer, but I realise I need the Forge server to install mods and plugins. 

Do anyone know how to safely change the type of the server without losing progress?

**Comment by itzg** [comment]
Just add the env var `TYPE` set to "FORGE". Nothing is lost from the world data as long as you're mounting the same data volume/directory.

**Reply by Tingweiftw** [reply]
Thanks! What is the safe docker command to run without deleting the data? Sorry I'm a docker noob

**Reply by itzg** [reply, marked as answer]
How are you creating your container now? In any case, please read https://docker-minecraft-server.readthedocs.io/en/latest/data-directory/

**Reply by Tingweiftw** [reply]
Thanks! I managed to reinstall forge without losing my world data by following 
```
docker run -d -it --name mc-new -v mc:/data -p 25565:25565 -e EULA=TRUE -e MEMORY=2G itzg/minecraft-server
```

Great work man thanks for the quick responses
